### PR TITLE
appdata -> metainfo, add support for metainfo files

### PR DESCRIPTION
**Pull request by Conan-Kudo**
These provides are specifically for packages providing AppStream files, which are either going to be `*.appdata.xml` or `*.metainfo.xml` files in `/usr/share/appdata` or `/usr/share/metainfo`.

The upstream AppStream specification mandates *.metainfo.xml files installed into `/usr/share/metainfo`, but there's still a large body of legacy AppStream files installed in the legacy location.

For now, let's support both under the new `metainfo()` Provides.

### CODE REVIEW

File: scripts/appstream.prov [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+OLD_IFS="$IFS"
+while read instfile ; do
+	case "$instfile" in
+	*.appdata.xml)
+		echo "appstream()"
+		echo "appstream(${instfile##*/appdata/})"
+		;;
+	*.metainfo.xml)
+		echo "appstream()"
+		echo "appstream(${instfile##*/metainfo/})"
```
Comment (ignatenkobrain) [inline scripts/appstream.prov]: I think for plugins it should not be same...

**Comment by mlschroe** [comment]
Wait a sec, those files are appdata, not appstream. Appstream is the data included in repositories. Appdata is the data installed in /usr/share/appdata and /usr/share/metainfo.

**Comment by Conan-Kudo** [comment]
@mlschroe That's not how @ximion explained it to me. He seemed to indicate that AppStream covers both aspects, and supersedes the old AppData spec.

File: scripts/appstream.prov (hunk 1 above)
Comment (Conan-Kudo) [inline scripts/appstream.prov]: `*.metainfo.xml` is being used for everything now. You can see this in new KDE AppStream files, and GNOME ones as well. They aren't called `*.appdata.xml` anymore.

**Comment by Conan-Kudo** [comment]
Hmm, apparently there is a difference... https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html

**Comment by ximion** [comment]
@Conan-Kudo I understand this confusion, as in the early days AppStream was indeed "what distros ship" and the upstream metainfo files followed later, being part of AppStream and making the whole naming a mess. Since naming is especially important for a spec, this stuff is now sorted out properly:
 * Data shipped by upstream projects is called *metainfo* and stored in `/usr/share/metainfo` (or the legacy location `/usr/share/appdata` - whether the files are named `.appdata.xml` or `.metainfo.xml` doesn't matter, but convention is to name the metainfo files for applications `.appdata.xml`
 * Data shipped by distributors is called *collection metadata* and stored in `/usr/(share|lib|cache)/app-info/<format>`

**Comment by Conan-Kudo** [comment]
@ximion Would `as-metainfo` work better as a name than `appstream` or `metainfo`? I want it to be clear what this actually is (a form of AppStream data).

**Comment by Conan-Kudo** [comment]
I think I want to just leave it as `appstream()`, as then we won't get bitten again by changes in what actually bits of AppStream are called.

**Comment by ximion** [comment]
The naming will never change again, unless we have a really really good reason for breakage (and I see none there).
`as-metainfo` would be fine if it fits the general style of these RPM names, `appstream` would also be okay - I don't really have preferences, take whatever you think is best.

**Comment by Conan-Kudo** [comment]
@ximion I've elected to change everything to `metainfo()`, as it resembles what we did before with calling them `appdata()`.

**Comment by ffesti** [comment]
Thanks for the patch! Merged.

**Comment by mlschroe** [comment]
How about we remove the appdata provides generator from rpm? It's not used anymore anyway. The reason I added it was so I did not need to fetch the complete package filelist, as /usr/share/appdata is not included in primary.xml. That's why the original generator just added some provides for the file names.

It turned out that this approach wasn't very useful, as some packages used pretty much random filenames instead if the ids. So IMHO scrap the thing. Or replace it with some code that parses the xml data.

**Comment by Conan-Kudo** [comment]
@mlschroe I may come back later to make the generator a little smarter, but I wanted the name/path fixed first, as I rely on it for stuff myself.

**Pull request closed without merging**